src/Tribe/Editor/Hooks.php

```diff
@@ -27,6 +30,40 @@ public function register() {
 	 */
 	protected function add_actions() {
 		add_action( 'current_screen', [ $this, 'add_widget_resources' ] );
+		add_action( 'save_post_' . TEC::POSTTYPE, [ $this, 'calculate_duration' ], 20 );
+	}
+
+	/**
+	 * Calculate the event duration based on the saved start and end dates.
+	 * This occurs during post save, and checks the saved value against the calculated value on updates.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $post_id The event we are setting a duration for.
+	 */
+	public function calculate_duration( $post_id ) {
+		// Sanity check.
+		if ( TEC::POSTTYPE !== get_post_type( $post_id ) ) {
+			return;
+		}
+
+		$saved_duration = get_post_meta( $post_id, '_EventDuration', true  );
+		$start_date_utc = get_post_meta( $post_id, '_EventStartDateUTC', true );
+		$end_date_utc   = get_post_meta( $post_id, '_EventEndDateUTC', true  );
```

Comment:
Working on the CT1 migration and related fixes, I found out that many sites will have varying combinations of localized start dates and UTC start dates; some due to being old, some due to plugins that will insert events directly in the database and will leave out meta we require. To deal with that, in CT1 code, [we use thist logic](https://github.com/the-events-calendar/the-events-calendar/blob/7179da28512f3bd23948f0a16f15510a04749588/src/Events/Custom_Tables/V1/Migration/Process_Worker.php#L847-L888) -- I think the best course of action here would be to either unify this logic and write it only once, or to make this logic more robust and be able to deal with mixed scenarios (e.g. `_EventStartDate` and `_EventEndDateUTC`).